cmd/ocitools/generate.go

```diff
@@ -305,61 +311,73 @@ func setupSpec(g *generate.Generator, context *cli.Context) error {
 		}
 	}
 
-	var sd string
-	var sa, ss []string
-
-	if context.IsSet("seccomp-default") {
-		sd = context.String("seccomp-default")
+	err := addSeccomp(g, context)
+	if err != nil {
+		return err
 	}
 
-	if context.IsSet("seccomp-arch") {
-		sa = context.StringSlice("seccomp-arch")
-	}
+	return nil
+}
 
-	if context.IsSet("seccomp-syscalls") {
-		ss = context.StringSlice("seccomp-syscalls")
+func addSeccomp(g *generate.Generator, context *cli.Context) error {
+
+	// Set the DefaultAction of seccomp
+	seccompDefault := context.String("seccomp-default")
+	err := g.SetDefaultSeccompAction(seccompDefault)
+	if err != nil {
+		return err
 	}
 
-	if sd == "" && len(sa) == 0 && len(ss) == 0 {
-		return nil
+	// Add the additional architectures permitted to be used for system calls
+	seccompArch := context.String("seccomp-arch")
+	architectureArgs := strings.Split(seccompArch, ",")
+	err = g.SetSeccompArchitectures(architectureArgs)
+	if err != nil {
+		return err
 	}
 
-	// Set the DefaultAction of seccomp
-	if context.IsSet("seccomp-default") {
-		if err := g.SetLinuxSeccompDefault(sd); err != nil {
+	if context.IsSet("seccomp-kill") {
+		seccompKill := context.String("seccomp-kill")
+		killArgs := strings.Split(seccompKill, ",")
+		err := g.SetSyscallActions("kill", killArgs)
```

Comment:
Not disagreeing, but what is the design advantage to passing the arguments one at a time instead of as a slice? 

Agreed on having a constant.

Again, if you can clarify about your idea for an option struct.